Task - 10 add image to Card

DmitryVDV opened the pull request:
1. Add image uploader via carrierwave
2. Add AWS S3 storage

DmitryVDV committed:
Task - 10 add image to Card

1. Add image uploader via carrierwave
2. Add AWS S3 storage

X1ting commented on .rubocop.yml:
```diff
@@ -94,19 +94,8 @@ Style/Documentation:
-    - 'app/models/user.rb'
-    - 'db/migrate/20170121125323_create_cards.rb'
-    - 'db/migrate/20170419071210_create_users.rb'
+    - 'app/**/*'
```
you really need rubocop, if you exclude all folders?

X1ting commented on app/models/card.rb:
```diff
@@ -1,8 +1,9 @@
+  validates :translated_text, :original_text, :cardimage, presence: { message: '!!!!!!!!!!!!!!!!!' }, on: :сreate
   validates :translated_text, exclusion: { in: :original_text }
   before_create :data_strip_downcase
+  mount_uploader :cardimage, CardimageUploader
```
why you call it cardimage, not just `image` ?

X1ting commented on app/models/card.rb:
```diff
@@ -1,8 +1,9 @@
 class Card < ApplicationRecord
   belongs_to :user
-  validates :translated_text, :original_text, presence: true
+  validates :translated_text, :original_text, :cardimage, presence: { message: '!!!!!!!!!!!!!!!!!' }, on: :сreate
```
what the strange message ?

X1ting replied:
and why on create olny?

X1ting commented on app/uploaders/cardimage_uploader.rb:
```diff
@@ -0,0 +1,52 @@
+    "flashcard777/flashcard777/uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  # Provide a default URL as a default if there hasn't been a file uploaded:
```
you need all this comments?

X1ting commented on Gemfile:
```diff
@@ -14,6 +14,9 @@ gem 'bootstrap-sass', '~> 3.2.0'
 gem 'simple_form'
+gem 'carrierwave', '~> 1.0'
+gem 'fog-aws'
+gem "mini_magick"
```
you use it?

X1ting commented on app/views/layouts/modal.html.erb:
```diff
@@ -0,0 +1,8 @@
+<!-- app/views/layouts/modal.html.erb -->
```
you need this comments?

X1ting commented on config/initializers/devise.rb:
```diff
@@ -108,7 +108,8 @@
   # Set up a pepper to generate the hashed password.
-  # config.pepper = '59fc887246272bee6da550543daa62f0095e2b80a5e392d715a9139ca2700559a2b52191d97afb214f5a6420b6535101e7a4fbb2671c06ea0e67b91633716ab2'
+  # config.pepper = '59fc887246272bee6da550543daa62f0095e2b80a5e392d715a9139c
+  # a2700559a2b52191d97afb214f5a6420b6535101e7a4fbb2671c06ea0e67b91633716ab2'
```
? do not add to commit usefull changes

X1ting reviewed (changes requested):
and add to .gitignore your images folder. They most not be in git index.

DmitryVDV committed:
Task - 10 for code review 09.06.17

1. made correction for card_controller’s logic when validation in cards
model are broken

2. correct CarrierWave config for storage. Insert it to
config/initializers

DmitryVDV committed:
Task 10 for code review 09.06.17

chang Cardimage column to Image

DmitryVDV committed:
Task-10 deleted unused comments

DmitryVDV committed:
Task-10 correct for hound

X1ting commented on app/interactors/card_review_date_three_days.rb:
```diff
@@ -2,13 +2,15 @@ class CardReviewDateThreeDays
     context.params[:review_date] = 3.days.since
 
-    if user.cards.create!(context.params)
+    newcard = user.cards.create(context.params)
```
better to use 'new_card' snail_case

X1ting commented on app/models/card.rb:
```diff
@@ -1,8 +1,9 @@
+  validates :translated_text, :original_text, :image, presence: true
   validates :translated_text, exclusion: { in: :original_text }
   before_create :data_strip_downcase
+  mount_uploader :image, CardimageUploader
```
CardImageUploader

X1ting commented on spec/features/card_image_spec.rb:
```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe CardimageUploader do
```
CardImage

X1ting commented on db/migrate/20170619170219_change_column_name.rb:
```diff
@@ -0,0 +1,5 @@
+class ChangeColumnName < ActiveRecord::Migration[5.0]
+  def change
+    rename_column :cards, :cardimage, :image
```
how you rename this column, if you not create it before?)

DmitryVDV committed:
Task-10 del image

X1ting commented on app/views/cards/index.html.erb:
```diff
@@ -15,6 +15,14 @@
               </div>
+
+              <div class="flex-item-other">
+                <%  if card.image.file.nil?  %>
```
strange condition. why not
```
if card.image?
  image_tag(card.image.small.url)
else 
  "Картинки нет"
end

X1ting commented on app/views/home/index.html.erb:
```diff
@@ -7,15 +7,18 @@
-      <div class="col-md-3">
+
+      <div class="col-md-4">
+        <%= image_tag(@cards.image.url) if @cards.image? %>
```
maybe @card here? Because you can not get .image, from @cards(collection)

DmitryVDV committed:
Task-10 code review 20.06.17
